server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -486,6 +486,10 @@ func (r *statsRecorder) Stop() {
 }
 
 func persistArtifact(ctx context.Context, env environment.Env, uri *url.URL, path string) error {
+	alreadyExists, err := env.GetBlobstore().BlobExists(ctx, path)
+	if err == nil && alreadyExists {
+		return nil
+	}
```

Comment:
this seems ok but not sure if we'll ever hit this case since I think we store these per-invocation?